superset-frontend/src/components/MetadataBar/dataset/DatasetMetadataBar.stories.tsx

```diff
@@ -0,0 +1,89 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { useResizeDetector } from 'react-resize-detector';
+import { css, SupersetClient } from '@superset-ui/core';
+import { useDatasetMetadataBar } from './useDatasetMetadataBar';
+
+export default {
+  title: 'Design System/Components/MetadataBar/Examples',
+  parameters: {
+    mockData: [
+      {
+        url: '/api/v1/dataset/1',
+        method: 'GET',
+        status: 200,
+        response: {
+          result: {
+            changed_on: '2023-01-26T12:06:58.733316',
+            changed_on_humanized: 'a month ago',
+            changed_by: { first_name: 'Han', last_name: 'Solo' },
+            created_by: { first_name: 'Luke', last_name: 'Skywalker' },
+            created_on: '2023-01-26T12:06:54.965034',
+            created_on_humanized: 'a month ago',
+            table_name: `This is dataset's name`,
+            owners: [
+              { first_name: 'John', last_name: 'Doe' },
+              { first_name: 'Luke', last_name: 'Skywalker' },
+            ],
+            description: 'This is a dataset description',
+          },
+        },
+      },
+    ],
+  },
+};
+
+export const DatasetSpecific = () => {
+  SupersetClient.reset();
```

Comment:
Can we mock `useApiV1Resource` instead of introducing a new dependency (`storybook-addon-mock`) and configuring `SupersetClient`?

Reply:
There's an [addon](https://storybook.js.org/addons/storybook-addon-manual-mocks) to mock imports but that would also introduce a new dependency. It seems it's not very straightforward though. While thinking about the problem, I'm not sure if a storybook for this hook is highly valuable given that the `MetadataBar` component already has one that displays the dataset information. I see a test file much for valuable as the objective of the hook is to assemble the correct information from the server and transform data to be displayed. If you asked me, I would delete the storybook and the new dependency and add a test file with the different combinations of server responses (no description, no creator, no owners, etc). I'll leave the decision to you as it's not a blocker for me.

Reply:
Added UT. Like you said, to mock a dependency we'd still need an addon.
I'd rather keep the addon for mocking API responses as I think it opens a possibility to write stories for more complex components and features.
However, I added UTs like you suggested